introduce the HDS graphics server deployable and docs

andponlin opened the pull request:
See the accompanying documentation `docs/service/haikudepotserver-server-graphics.md` for an overview. This PR will introduce a new K8S `deployment` and `service` called `haikudepotserver-server-graphics`. This PR does not update the HDS deployment instructions in `standard-change--haikudepot-deployment.md` or configure HDS to use the new service; this first step is to just get the new service running.

kallisti5 commented:
ooh. Nice!  Is there a substantial memory usage increase with this new service?   HDS is using around 700MiB when running.  

Here's our node resource utilization:
* CPU: 37%   Memory:   5.2GiB   78%
* CPU: 17%   Memory:   4.8GiB   71%
* CPU:   9%   Memory:   5.3GiB   79%

We have been running pretty tight for a while, so just want to get a handle on if we need to add another node to the cluster.

(full disclosure, HDS is actually pretty awesome on memory usage. The top consumers are discourse at 3.6GiB, Gerrit at 2.1GiB, and postgres at 2.5GiB)

andponlin commented:
Hello @kallisti5 ; When I bring HDS up in a profiler, it's baseline is about 80MB. If I run something like an HPKR import process, the process heap seems to float between 60 and 120MB. The Java process has it's heap specified in the container startup arguments with a lower-bound of 320MB and an upper bound of 512MB so a fixed memory allocation for the pod of 750MB is probably fine.

The new application server, `haikudepotserver-server-graphics` when profiled uses 30MB but I have a pinned memory heap size of 196MB which is actually probably too much but that's baked into the current `1.0.169` image. It would probably be OK to adjust the max pod memory to 320MB. It does use some transient `/tmp` storage but the pod's own `/tmp` is probably enough.

Actions;
- I have adjusted the HDS memory from 0.5 - 2GB to be 0.75GB
- I have adjusted the HDS-GS memory from 0.5GB to 320MB.

kallisti5 commented:
Thanks for looking at this!   I think we can probably squeeze this in without adding nodes.  We're definitely getting close to needing 4 though :sweat:

kallisti5 commented:
Deployed.  Things seem to be working as expected...

```
$ curl -o output-file.png -v -X POST --data-binary @input-file.png "http://localhost:9085/__gfx/thumbnail?w=100&h=100"
Note: Unnecessary use of -X or --request, POST is already inferred.
* Host localhost:9085 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying [::1]:9085...
* Connected to localhost (::1) port 9085
* using HTTP/1.x
> POST /__gfx/thumbnail?w=100&h=100 HTTP/1.1
> Host: localhost:9085
> User-Agent: curl/8.12.1
> Accept: */*
> Content-Length: 4852
> Content-Type: application/x-www-form-urlencoded
> 
} [4852 bytes data]
* upload completely sent off: 4852 bytes
< HTTP/1.1 200 
< Content-Type: image/png
< Transfer-Encoding: chunked
< Date: Sun, 02 Mar 2025 21:30:25 GMT
< 
{ [1089 bytes data]
100  5929    0  1077  100  4852    905   4079  0:00:01  0:00:01 --:--:--  4986
* Connection #0 to host localhost left intact
```

kallisti5 commented:
![Screenshot From 2025-03-02 15-32-45](https://github.com/user-attachments/assets/b6997602-b139-427d-88b4-5a317958ecce)

andponlin commented:
Thanks!
